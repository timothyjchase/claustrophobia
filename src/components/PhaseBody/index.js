import React from 'react'
import PropTypes from 'prop-types'
import InitiativePhaseContainer from '../../containers/InitiativePhaseContainer'
import HumanActionPhaseContainer from '../../containers/HumanActionPhaseContainer'
import ThreatPhaseContainer from '../../containers/ThreatPhaseContainer'
import DemonActionPhaseContainer from '../../containers/DemonActionPhaseContainer'

const PHASES = {
  INITIATIVE: InitiativePhaseContainer,
  HUMAN_ACTION: HumanActionPhaseContainer,
  THREAT: ThreatPhaseContainer,
  DEMON_ACTION: DemonActionPhaseContainer,
}

const PhaseBody = ({ phase }) => {
  const BodyComponent = PHASES[phase]
  return <BodyComponent />
}

PhaseBody.propTypes = {
  phase: PropTypes.string.isRequired,
}

export default PhaseBody
