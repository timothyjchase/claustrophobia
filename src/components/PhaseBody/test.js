import React from 'react'
import PhaseBody from './index'

describe('<PhaseBody />', () => {
  describe('with phase = INITIATIVE', () => {
    const component = shallow(<PhaseBody phase="INITIATIVE" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with phase = HUMAN_ACTION', () => {
    const component = shallow(<PhaseBody phase="HUMAN_ACTION" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with phase = THREAT', () => {
    const component = shallow(<PhaseBody phase="THREAT" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
  describe('with phase = DEMON_ACTION', () => {
    const component = shallow(<PhaseBody phase="DEMON_ACTION" />)

    it('should match snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
