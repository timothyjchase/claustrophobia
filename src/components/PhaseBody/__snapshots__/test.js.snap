// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhaseBody /> with phase = DEMON_ACTION should match snapshot 1`] = `<Connect(InitiativePhaseContainer) />`;

exports[`<PhaseBody /> with phase = HUMAN_ACTION should match snapshot 1`] = `<Connect(HumanActionPhaseContainer) />`;

exports[`<PhaseBody /> with phase = INITIATIVE should match snapshot 1`] = `<Connect(InitiativePhaseContainer) />`;

exports[`<PhaseBody /> with phase = THREAT should match snapshot 1`] = `<Connect(ThreatPhaseContainer) />`;
